crates/release_plz_core/src/release.rs

```diff
@@ -131,9 +140,41 @@ fn release_package(
         info!("published {}", package.name);
     }
 
+    if input.gh_release {
+        publish_release(git_tag(&package.name, &package.version.to_string()), input);
+    }
```

Comment:
can you move this block inside the `else` above?
We don't want to do a GitHub release if the user passed the `dry_run` flag.

This is how it should look:

```rust
    } else {
        wait_until_published(index, package)?;

        let git_tag = git_tag(&package.name, &package.version.to_string());
        repo.tag(&git_tag)?;
        repo.push(&git_tag)?;

        info!("published {}", package.name);

        if input.gh_release {
            publish_release(git_tag, input).await;
        }
    }
```
Notice that I reused the `git_tag` variable instead of calculating the git tag again :)